### PR TITLE
Draft: Clarifying local development installation in user guide

### PR DESCRIPTION
**Pull request by freddyaboulton**
##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by fkiraly** [comment]
So, @freddyaboulton, there is a duplication of these instructions apparently:

* in the `contributing.md`
* here, imo the best version: https://www.sktime.org/en/latest/installation.html

I think there are also links missing to the installation instructions:
* here: https://www.sktime.org/en/latest/developer_guide.html
* here: https://www.sktime.org/en/latest/get_started.html
* in the readme.md, in the "documentation" table at the top, imo

Now I'm not sure what the best way to resolve this is (e.g., remove material, link to one place).
But while you're at it, would be great if you could also have a look and possibly clean up this small mess in passing...

File: setup.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "scikit-learn": "0.24.0",
     "statsmodels": "0.12.1",
     "numba": "0.53",
+    "esig": "0.9.7",
```
Comment (fkiraly) [inline setup.py:29]: this should go to all extras and `esig` a soft dependency, no?

**Comment by fkiraly** [comment]
I think this is superseded by https://github.com/alan-turing-institute/sktime/pull/1672?

**Comment by freddyaboulton** [comment]
Thanks for closing @fkiraly

**Pull request closed without merging**